src/repositories/prisma/prisma-users-repository.ts

```diff
@@ -0,0 +1,25 @@
+import { Prisma } from '@prisma/client';
+import { prisma } from '../../lib/prisma';
+import { UserRepository } from '../user-repository';
+
+export class PrismaUsersRepository implements UserRepository {
+  async findByEmail(email: string) {
+    const user = await prisma.user.findUnique({
+      where: {
+        email,
+      },
+    })
+
+    return user
+  }
+  async findById(id: string): Promise<{ id: string; name: string; surname: string; email: string; password_hash: string; created_at: Date; updated_at: Date; } | null> {
```

Comment:
Hey, could you follow just one pattern for the return types? For **findByEmail**, you didn't specify the return type.

Suggestion: You can add return types to both methods to make the repository easier to understand. You can follow the return types that are in the interface at user-repository.ts.